### PR TITLE
Two builds, one repository: chefdk build via WS

### PR DESCRIPTION
**Pull request by marcparadise**
## Description

Experimental support for building both chef-dk
and chef-workstation.  This commit adds a chefdk project to
chef-workstation's omnibus configuration, and brings
chef-workstation's software definitions up to date with chef-dk.

This does not address pipeline changes - it only provides a means to
build both packages from the chef-workstation repository.

 It also doesn't attempt to do any cleanup - it's just taking the build and 
transplanting it, with the  intent of having the chef-dk repository serve as 
the repository for only the dk gem. 

I've validated this for DK and WS omnibus build, eg
`omnibus build chefdk` and `omnibus build chef-workstation`.

The former works and generates a DK package. The latter fails in the
electron setup for chef-workstation-app - I don't think this is related
but will dig further.

Signed-off-by: Marc A. Paradise <marc.paradise@gmail.com>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: omnibus/config/projects/chefdk.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+end
+
+package :msi do
+  fast_msi true
+  upgrade_code "AB1D6FBD-F9DC-4395-BDAD-26C4541168E7"
+  signing_identity "E05FF095D07F233B78EB322132BFF0F035E11B5B", machine_store: true
+  wix_light_extension "WixUtilExtension"
+end
+
+# We don't support appx builds, and they eat a lot of time.
```
Comment (marcparadise) [inline omnibus/config/projects/chefdk.rb]: This is new for DK.

File: omnibus/config/software/ruby-windows-system-libraries.rb [hunk 2]
```diff
@@ -14,8 +14,12 @@
     dlls = [
       "libwinpthread-1",
       "libstdc++-6",
-      "libgcc_s_seh-1",
     ]
+    if windows_arch_i386?
```
Comment (marcparadise) [inline omnibus/config/software/ruby-windows-system-libraries.rb]: This seemed like a thing that chef-workstation should also have.

File: omnibus/kitchen.yml [hunk 3]
```diff
@@ -26,12 +26,17 @@ platforms:
 
 suites:
   - name: default
```
Comment (marcparadise) [inline omnibus/kitchen.yml]: Temporary for manual runs

File: omnibus/config/projects/chefdk.rb [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name "chefdk"
+friendly_name "Chef Development Kit"
+maintainer "Chef Software, Inc. <maintainers@chef.io>"
+homepage "https://www.chef.io"
+license "Chef EULA"
```
Comment (marcparadise) [inline omnibus/config/projects/chefdk.rb]: Should chef-workstation use this as well? It's currently using Apache 2 for the project.

File: omnibus/CHEF-EULA.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+
```
Comment (marcparadise) [inline omnibus/CHEF-EULA.md]: Copied in unmodified from chef-dk/omnibus/CHEF-EULA.md

**Comment by tas50** [comment]
Closing this out since we're going to pull chef-cli gem in an manage our own gemfile here insead

**Pull request closed without merging**